src/stability_sdk/client.py

```diff
@@ -104,30 +119,147 @@ def process_artifacts_from_answers(
             idx += 1
 
 
-def open_channel(host: str, api_key: str = None) -> generation_grpc.GenerationServiceStub:
-    print(f"Opening channel {host}")
-    if host.endswith(":443"):
-        call_credentials = [grpc.access_token_call_credentials(api_key)]
-        channel_credentials = grpc.composite_channel_credentials(
-            grpc.ssl_channel_credentials(), *call_credentials
-        )
-        channel = grpc.secure_channel(host, channel_credentials)
-    else:
-        channel = grpc.insecure_channel(host)
-    return generation_grpc.GenerationServiceStub(channel)
-
 
 class ApiEndpoint:
     def __init__(self, stub, engine_id):
         self.stub = stub
         self.engine_id = engine_id
 
+class Project():
+    def __init__(self, api: 'Api', project: project.Project):
+        self.api = api
+        self.id = project.id
+        self.file_id = project.file.id
+        self.title = project.title
+
+    @staticmethod
+    def create(
+        api: 'Api', 
+        title: str, 
+        access: project.ProjectAccess=project.PROJECT_ACCESS_PRIVATE,
+        status: project.ProjectStatus=project.PROJECT_STATUS_ACTIVE
+    ) -> 'Project':
+        req = project.CreateProjectRequest(title=title, access=access, status=status)
+        proj: project.Project = api._proj_stub.Create(req, wait_for_ready=True)
+        return Project(api, proj)
+
+    def delete(self):
+        self.api._proj_stub.Delete(project.DeleteProjectRequest(id=self.id))
+
+    @staticmethod
+    def list_projects(api: 'Api') -> List['Project']:
+        list_req = project.ListProjectRequest(owner_id="")
+        results = []
+        for proj in api._proj_stub.List(list_req, wait_for_ready=True):
+            results.append(Project(api, proj))
+        return results
+
+    def load_settings(self) -> dict:
```

Comment:
following the 'property' pattern I recommended above, you could add "settings" as a lazily loaded attribute on the class like this:

```python
    ...
    @property
    def settings(self):
        if not hasattr(self, '_settings'):
            self._settings = self.load_settings()
        return self._settings
```

Reply:
I would like to think more about this and consider for a future change. I think it is good to be an explicit method to load the file and caller should handle any exception from the RPC. If it is wrapped as a property it could encourage being accessed anywhere and lead to an exception in unexpected places. That's my thought right now at least and as project support is developed out more it may be more clear one way or the other...

Reply:
yeah can def revisit this later. the model I have in my mind rn is the reddit api. their sdk is a python library called PRAW which has a lot of extremely helpful lazily evaluated attributes on its classes that abstracts a lot of usage complexity away from the user. For example, comment threads on reddit are nested and have unbounded depth, but you can traverse them naturally with PRAW objects which only expand the comment tree incrementally as needed. Maybe we don't need settings to be populated this way, but I think this is potentially a generally useful paradigm for interacting with projects, where neither the user nor the server will want to dump all of the information/artifacts at once